python/tank/descriptor/io_descriptor/base.py

```diff
@@ -497,27 +497,18 @@ def dict_from_uri(cls, uri):
         #
         # uri: sgtk:descriptor:app_store?version=v0.1.2&name=tk-bundle
         #
-        # python 2.6+ expected: ParseResult(
+        # python expected: ParseResult(
         # scheme='sgtk',
         # netloc='',
         # path='descriptor:app_store',
         # params='',
         # query='version=v0.1.2&name=tk-bundle',
         # fragment='')
-        #
-        # python 2.5 and others: (
-        # 'sgtk',
-        # '',
-        # 'descriptor:app_store?version=v0.1.2&name=tk-bundle',
-        # '',
-        # '',
-        # '')
 
         if parsed_uri.scheme != constants.DESCRIPTOR_URI_PATH_SCHEME:
             raise TankDescriptorError("Invalid uri '%s' - must begin with 'sgtk'" % uri)
 
         if parsed_uri.query == "":
-            # in python 2.5 and others, the querystring is part of the path (see above)
             (path, query) = parsed_uri.path.split("?")
         else:
             path = parsed_uri.path
```

Comment:
I would be surprised if this bug still exists. I think you can remove the condition